pkg/network/common/egressip.go

```diff
@@ -420,43 +421,41 @@ func (eit *EgressIPTracker) syncEgressNodeState(eg *egressIPInfo, active bool) {
 
 func (eit *EgressIPTracker) syncEgressNamespaceState(ns *namespaceEgress) {
 	if len(ns.requestedIPs) == 0 {
-		if ns.activeEgressIP != "" {
-			ns.activeEgressIP = ""
+		if ns.activeEgressIP != nil {
+			ns.activeEgressIP = nil
 			eit.watcher.SetNamespaceEgressNormal(ns.vnid)
 		}
 		return
 	}
 
-	var active *egressIPInfo
+	activeEgressIPs := make(map[string]string)
 	for _, ip := range ns.requestedIPs {
 		eg := eit.egressIPs[ip]
 		if eg == nil {
 			continue
 		}
 		if len(eg.namespaces) > 1 {
-			active = nil
 			klog.V(4).Infof("VNID %d gets no egress due to multiply-assigned egress IP %s", ns.vnid, eg.ip)
+			activeEgressIPs = nil
 			break
 		}
-		if active == nil {
-			if eg.assignedNodeIP == "" {
-				klog.V(4).Infof("VNID %d cannot use unassigned egress IP %s", ns.vnid, eg.ip)
-			} else if len(ns.requestedIPs) > 1 && eg.nodes[0].offline {
-				klog.V(4).Infof("VNID %d cannot use egress IP %s on offline node %s", ns.vnid, eg.ip, eg.assignedNodeIP)
-			} else {
-				active = eg
-			}
+		if eg.assignedNodeIP == "" {
+			klog.V(4).Infof("VNID %d cannot use unassigned egress IP %s", ns.vnid, eg.ip)
+		} else if len(ns.requestedIPs) > 1 && eg.nodes[0].offline {
+			klog.V(4).Infof("VNID %d cannot use egress IP %s on offline node %s", ns.vnid, eg.ip, eg.assignedNodeIP)
+		} else {
+			activeEgressIPs[eg.ip] = eg.assignedNodeIP
 		}
 	}
 
-	if active != nil {
-		if ns.activeEgressIP != active.ip {
-			ns.activeEgressIP = active.ip
-			eit.watcher.SetNamespaceEgressViaEgressIP(ns.vnid, active.ip, active.assignedNodeIP)
+	if len(activeEgressIPs) > 0 {
+		if !reflect.DeepEqual(ns.activeEgressIP, activeEgressIPs) {
+			ns.activeEgressIP = activeEgressIPs
+			eit.watcher.SetNamespaceEgressViaEgressIP(ns.vnid, activeEgressIPs)
 		}
 	} else {
-		if ns.activeEgressIP != "dropped" {
-			ns.activeEgressIP = "dropped"
+		if ns.activeEgressIP["dropped"] != "dropped" {
+			ns.activeEgressIP = map[string]string{"dropped": "dropped"}
```

Comment:
meh, this was an ugly hack before and it's an uglier hack now. should probably have a separate field for this... 

(The important thing going on here is that we need to be able to distinguish "there are `requestedIPs` but no `activeEgressIPs` and we haven't called `SetNamespaceEgressDropped` yet" from "there are `requestedIPs` but no `activeEgressIPs` and we _have_ already called `SetNamespaceEgressDropped`", because we don't want to call it again every time this function runs.)

Reply:
done